app/mm-redux/utils/post_list.ts

```diff
@@ -97,7 +97,7 @@ function selectOrderedPostIds(posts: types.posts.Post[], lastViewedAt: number, i
 
         if (
             lastViewedAt &&
-            post.create_at > lastViewedAt &&
+            post.create_at >= lastViewedAt &&
```

Comment:
will this still work in channels?

Reply:
I don't see why not

Reply:
It depends when marking something as unread, do we set the last viewed at to be `post.create_at` or `post.create_at + 1` I guess.

Reply:
We are setting it to `Date.now()` on marking a thread as read.

For marking as unread:
1. Posts,  we are doing `create_at - 1` (We are just calling the api and api value is this)
2. Threads, we are sending `create_at` to the server.

Reply:
Sorry I deleted the message @anurag6713, what happens when we are marking a channel as read?

Reply:
We are calling `set_unread` api with channel id.

However for the thread posts, we are calling `thread read` api with the `timestamp`.

Reply:
It seems that it is set to `lastPostAt` value.
https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/channel_store.go#L2147

If I'm correct if you mark the last in a channel as unread you then could never clear it being unread, can you either verify or refute this hypothesis?

Reply:
You are right @koox00 .. Thank you 🙂 Reverted the code and reducing the value by `1` to make It work for the threads.